rustybeer-cli/src/commands/alcohol_volume_weight.rs

```diff
@@ -0,0 +1,101 @@
+use clap::{value_t, App, Arg, ArgMatches};
+pub use rustybeer::calculators::alcohol_volume_weight::{
+    calculate_abv_abw, calculate_abv_abw_density, calculate_abw_abv, calculate_abw_abv_density,
+    calculate_alc_vol, calculate_alc_weight,
+};
+use rustybeer_util::conversions::VolumeBuilder;
+
+pub fn add_subcommand<'a, 'b>() -> App<'a, 'b> {
+    App::new("abv_abw")
+        .version("0.1")
+        .author("Joseph Russell (josephrussell123@gmail.com)")
+        .about("Calculates Alcohol by Weight (ABW) from Alcohol By Volue (ABV) and vice versa")
+        .arg(
+            Arg::with_name("percent")
+                .short("p")
+                .long("percent")
+                .value_name("PERCENT")
+                .help("'From' alcohol percentage")
+                .required(true)
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("total_volume")
+                .short("v")
+                .long("total_volume")
+                .value_name("TOTAL_VOLUME")
+                .help("Total beer volume")
+                .required(false)
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("total_density")
+                .short("d")
+                .long("total_density")
+                .value_name("TOTAL_DENSITY")
+                .help("Total density of beer in g/cm³")
+                .required(false)
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("reverse")
+                .short("r")
+                .long("reverse")
+                .value_name("REVERSE")
+                .help("Calculates ABW to ABV")
+                .required(false)
+                .takes_value(false),
+        )
+}
+
+pub fn do_matches(matches: &ArgMatches) {
+    if let Some(matches) = matches.subcommand_matches("abv_abw") {
+        if matches.is_present("percent") {
+            let percent = value_t!(matches, "percent", f64).unwrap_or_else(|e| e.exit());
+            let end_percentage: f64;
+
+            // main ABV <-> ABW conversion
+            if matches.is_present("reverse") {
+                if matches.is_present("total_density") {
+                    let total_density =
+                        value_t!(matches, "total_density", f64).unwrap_or_else(|e| e.exit());
```

Comment:
As said earlier, you could create DensityBuilder to convert from string. But that can be done later or when the measurements library supports FromStr trait